### PR TITLE
feat: Allow users to mark a conversation as unread

### PR DESCRIPTION
**Pull request by sojan-official**
fixes: #5552 

Loom video: https://www.loom.com/share/ab70552d3c9c48b685da7dfa64be8bb3

### CODE REVIEW

**Comment by pranavrajs** [comment]
@muhsin-k @fayazara @iamsivin Please review the PR and test out the functionality. 

@tejaswinichile Please take a look at the backend changes.

File: app/javascript/dashboard/store/modules/conversations/actions/messageReadActions.js [hunk 1]
```diff
@@ -0,0 +1,35 @@
+import { throwErrorMessage } from 'dashboard/store/utils/api';
```
Comment (pranavrajs) [inline app/javascript/dashboard/store/modules/conversations/actions/messageReadActions.js:1]: Created a new file as the action.js has passed 300 lines. Modularizing parts of the code would be required.

**Review by fayazara — APPROVED** [review]
FE code looks.

Tested locally - working as expected.

https://www.loom.com/share/d4aa13d885de462883ed89cf7788749a

**Review by muhsin-k — APPROVED** [review]

**Comment by nithindavid** [comment]
Should we keep a button next to the resolve button(adjacent to mute) to toggle the same and with a tooltip?

**Comment by tejaswinichile** [comment]
The backend looks good, why did you add skipModelValidation in robocop?

**Comment by sojan-official** [comment]
@tejaswinichile we are doing an update column here, which rubocop doesn't like. but it is intended in this case, as we don't want the callbacks.

**Comment by sojan-official** [comment]
> Should we keep a button next to the resolve button(adjacent to mute) to toggle the same and with a tooltip?

We could take it as a subsequent task if there is enough usage for this functionality. The current customer requirement only wanted it in the context menu.  will convert it to a follow up issue.

**Comment by pranavrajs** [comment]
@nithindavid This could be a good enhancement. Let us rework on the top panel and place important items there.

**Pull request merged**